azure-quantum/azure/quantum/target/microsoft/result.py

```diff
@@ -0,0 +1,237 @@
+##
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+##
+from typing import Any, Dict, List, Optional, Union
+
+import markdown
+
+
+class HTMLWrapper:
+    """
+    Simple HTML wrapper to expose _repr_html_ for Jupyter clients.
+    """
+    def __init__(self, content: str):
+        self.content = content
+
+    def _repr_html_(self):
+        return self.content
+
+
+class MicrosoftEstimatorResult(dict):
+    MAX_DEFAULT_ITEMS_IN_TABLE = 5
+
+    def __init__(self, data: Union[Dict, List]):
+        self._data = data
+
+        if isinstance(data, dict):
+            super().__init__(data)
+
+            self._is_simple = True
+            self._repr = _item_result_table(self)
+            self.summary = HTMLWrapper(_item_result_summary_table(self))
+        elif isinstance(data, list):
+            super().__init__({idx: MicrosoftEstimatorResult(item_data) for idx, item_data in enumerate(data)})
+
+            self._data = data
+            self._is_simple = False
+            num_items = len(data)
+            self._repr = ""
+            if num_items > self.MAX_DEFAULT_ITEMS_IN_TABLE:
+                self._repr += f"<p><b>Info:</b> <i>The overview table is cut off after {self.MAX_DEFAULT_ITEMS_IN_TABLE} items.  If you want to see all items, suffix the result variable with <code>[:]</code></i></p>"
+                num_items = self.MAX_DEFAULT_ITEMS_IN_TABLE
+            self._repr += _batch_result_table(self, range(num_items))
+
+    def data(self, idx: Optional[int] = None) -> Any:
+        """
+        Returns raw data of the result object.
+
+        In case of a batching job, you can pass an index to access a specific
+        item.
+        """
+        if idx is None:
+            return self._data
+        elif not self._is_simple:
+            return self._data[idx]
+        else:
+            raise ValueError("Cannot pass parameter 'idx' to 'data' for non-batching job")
+
+    def _repr_html_(self):
+        """
+        HTML table representation of the result.
+        """
+        return self._repr
+
+    def __getitem__(self, key):
+        """
+        If the result represents a batching job and key is a slice, a
+        side-by-side table comparison is shown for the indexes represented by
+        the slice.
+
+        Otherwise, the key is used to access the raw data directly.
+        """
+        if isinstance(key, slice):
+            if self._is_simple:
+                raise ValueError("Cannot pass slice to '__getitem__' for non-batching job")
+            return HTMLWrapper(_batch_result_table(self, range(len(self))[key]))
+        else:
+            return super().__getitem__(key)
+
+def _item_result_table(result):
```

Comment:
Shouldn't `_item_result_table`, `_item_result_summary_table` and `_batch_result_table` methods be inside the `MicrosoftEstimatorResult`?

Reply:
Yes, I'll move them inside.